server/tests/src/test/java/org/infinispan/server/persistence/JdbcStringBasedCacheStorePassivation.java

```diff
@@ -0,0 +1,234 @@
+ * <p>
+ * passivation == false --all cache entries should always be also in the cache store
+ * preload == true --after server restart, all entries should appear in the cache immediately
+ * purge == false --all entries should remain in the cache store after server  * restart
+ * <p>
+ * Other attributes like singleton, shared, fetch-state do not make sense in single node cluster.
+ *
+ */
+@Category(Persistence.class)
+@RunWith(Parameterized.class)
+public class JdbcStringBasedCacheStorePassivation {
+
+    @ClassRule
+    public static InfinispanServerRule SERVERS = PersistenceIT.SERVERS;
+
+    @Rule
+    public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+    private final Database database;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        String[] databaseTypes = PersistenceIT.DATABASE.getDatabaseTypes();
+        List<Object[]> params = new ArrayList<>(databaseTypes.length);
+        for (String databaseType : databaseTypes) {
+            params.add(new Object[]{databaseType});
+        }
+        return params;
+    }
+
+    public JdbcStringBasedCacheStorePassivation(String databaseType) {
+        this.database = PersistenceIT.DATABASE.getDatabase(databaseType);
+    }
+
+    @Test(timeout = 600000)
+    public void testFailoverWithPassivationDisabled() throws Exception {
+        JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC, database, false, true)
+                .setLockingConfigurations();
+        RemoteCache<String, String> cache = SERVER_TEST.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+        try(TableManipulation table = new TableManipulation(cache.getName(), jdbcUtil.getPersistenceConfiguration(), jdbcUtil.getConfigurationBuilder())) {
+            cache.put("k1", "v1");
+            cache.put("k2", "v2");
+            // test passivation==false, database should contain all entries which are in the cache
+            assertNotNull(table.getValueByKey("k1"));
+            assertNotNull(table.getValueByKey("k2"));
+
+            SERVERS.getServerDriver().stop(0);
+            SERVERS.getServerDriver().restart(0);
+
+            assertNotNull(table.getValueByKey("k1"));
+            assertNotNull(table.getValueByKey("k2"));
+            assertNull(cache.withFlags(Flag.SKIP_CACHE_LOAD).get("k3"));
+            assertEquals("v1", cache.get("k1"));
+            assertEquals("v2", cache.get("k2"));
+            //when the entry is removed from the cache, it should be also removed from the cache store (the store
+            //and the cache are the same sets of keys)
+            cache.remove("k1");
+            assertNull(table.getValueByKey("k1"));
+            cache.clear();
+        }
+
+    }
+
+    @Test(timeout = 600000)
+    public void testPreloadWithoutPurgePassivationFalse() throws Exception {
```

Comment:
The default behaviour is for `purge = false; passivation = false`, so I think `testPreload()` is sufficient.